### PR TITLE
MKc9LBJA: Add hubConnectorEntityId into HubEnvironment Enum

### PR DESCRIPTION
**Pull request by kerrr**
Added a HubEnvironment entry for hubConnectorEntityId, used by eidas and
default to using it when there is not hubConnectorEntityId specified in
the verify-service-provider.yml config.

co-authored-by: Olakunle Jegede <olakunle.jegede@digital.cabinet-office.gov.uk>

### CODE REVIEW

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java [hunk 1]
```diff
@@ -16,13 +16,13 @@
 public class EuropeanIdentityConfiguration extends EidasMetadataConfiguration {
 
     private TrustStoreConfiguration trustStoreConfiguration;
-    private String hubConnectorEntityId;
+    private URI hubConnectorEntityId;
```
Comment (jakubmiarka) [inline src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: It doesn't have to be necessarily URI I believe 🤔

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/ConfigurationConstants.java [hunk 2]
```diff
@@ -19,26 +19,35 @@
...
+    String METADATA_PATH = "/SAML2/metadata/federation";
+    String TRUSTANCHOR_PATH = "/SAML2/metadata/trust-anchor";
+    String METADATASOURCE_PATH = "/SAML2/metadata/aggregator";
+    String HUBCONNECTOR_ENTITY_ID_PATH = "/SAML2/metadata/connector";
+
+    String PRODUCTION_SSO = PRODUCTION_HOST + SSO_PATH;
+    String PRODUCTION_METADATA = PRODUCTION_HOST + METADATA_PATH;
+    String PRODUCTION_TRUSTANCHOR_URI = PRODUCTION_HOST + TRUSTANCHOR_PATH;
+    String PRODUCTION_METADATASOURCE_URI = PRODUCTION_HOST + METADATASOURCE_PATH;
+    String PRODUCTION_HUBCONNECTOR_ENTITYID = PRODUCTION_HOST + HUBCONNECTOR_ENTITY_ID_PATH;
```
Comment (jakubmiarka) [inline src/main/java/uk/gov/ida/verifyserviceprovider/configuration/ConfigurationConstants.java]: can I be really picky and say `PRODUCTION_HUBCONNECTOR_ENTITYID` should be `PRODUCTION_HUBCONNECTOR_ENTITY_ID`

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java (hunk 1 above)
Comment (kerrr) [reply src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: Hurm, I think in SAML the entityIds are all URIs 🤔

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/ConfigurationConstants.java (hunk 2 above)
Comment (kerrr) [reply src/main/java/uk/gov/ida/verifyserviceprovider/configuration/ConfigurationConstants.java]: Pick away ;)

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java (hunk 1 above)
Comment (jakubmiarka) [reply src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: Yep, you might be right!

Comment (kerrr) [reply src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: OK, higher powers agree with you :)

File: src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java [hunk 3]
```diff
@@ -53,6 +49,10 @@ public boolean isEnabled() {
         return enabled;
     }
 
+    public String getHubConnectorEntityId() {
```
Comment (chrisholmes) [inline src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: Could we either use Optional here or switch the other methods to do null checks in a consistent way?

Comment (kerrr) [reply src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java]: OK

**Comment by kerrr** [comment]
[Use hub.environment value to configure hub connector entity ID eIDAS field](https://trello.com/c/MKc9LBJA/195-use-hubenvironment-value-to-configure-hub-connector-entity-id-eidas-field)

**Pull request merged**